crml/governance/src/lib.rs

```diff
@@ -286,4 +371,64 @@ impl<T: Config> Module<T> {
 	pub fn get_proposal_votes() -> Vec<(ProposalId, ProposalVoteInfo)> {
 		ProposalVotes::iter().collect()
 	}
+
+	pub fn check_voter_account_validity(account: &T::AccountId) -> DispatchResult {
+		// Check the amount they have staked
+		let staked_amount: Balance = T::StakingInfo::active_balance(account.clone());
+		ensure!(staked_amount > 0, Error::<T>::NotEnoughStaked);
+
+		// Check their verified identities
+		// let registration: u32 = T::Registration::registered_accounts(account.clone());
+		// ensure!(
+		// 	registration >= MINIMUM_REGISTERED_IDENTITIES,
+		// 	Error::<T>::NotEnoughRegistrations
+		// );
+		Ok(())
+	}
+
+	pub fn end_referendum(proposal_id: ProposalId) {
+		let proposal = match Self::proposals(proposal_id) {
+			Some(proposal) => proposal,
+			None => {
+				warn!("clean up proposal: {:?} failed, not found", proposal_id);
+				return;
+			}
+		};
+		let mut no_vote_weight_sum: u64 = 0;
+		let max_stakers: u64 = T::StakingInfo::count_nominators();
+		// Sum up total vote weight
+		<ReferendumVoteCount<T>>::drain_prefix(proposal_id).for_each(|(_, vote)| {
+			no_vote_weight_sum += vote.vote as u64;
+		});
```

Comment:
use a tracker to keep count of votes as they're cast
and this will become smth like `<ReferendumVoteCount<T>>::kill_prefix(proposal_id)`
which is cheaper